.github/workflows/code_checks.yml

```diff
@@ -49,3 +49,22 @@ jobs:
       - uses: actions/checkout@v1
       - run: pip install -r ./.github/scripts/requirements.txt
       - run: ./.github/scripts/codecheck.py
+
+  cargo_deny_checks:
+    name: Cargo deny check licenses and advisories
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        checks:
+          - advisories
+          - licenses
+
+    # Prevent sudden announcement of a new advisory from failing ci:
+    continue-on-error: ${{ matrix.checks == 'advisories' }}
+
+    steps:
+    - uses: actions/checkout@v3
```

Comment:
Btw this needs some formatting to match the others. My IDE suggested that.

Reply:
Thanks for the comment. Do you mean with regards the indentation format?

Reply:
Yes

Reply:
Corrected. Thanks! Let me know if you need me to rebase it and have less commits

Reply:
I would prefer a rebase. But let's wait until we resolve the CI issue. You'll have to rebase then to include the CI changes so that we can built your PR. We're working on that.